File: baekjoon/해시/11652.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Map<Long, Integer> note = new HashMap<>();
        Long max_v = -(2L << 62) - 1L;
        int max_cnt = 0;

        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; ++i) {
            Long v = Long.parseLong(br.readLine());
            note.put(v, note.getOrDefault(v, 0) + 1);
            if (note.get(v) > max_cnt) {
                max_cnt = note.get(v);
                max_v = v;
            } else if (note.get(v) == max_cnt) {
                max_v = Math.min(max_v, v);
            }
        }

        bw.write(max_v.toString());
        bw.write('\n');
        bw.flush();
        bw.close();
    }
}

File: baekjoon/해시/11652.cpp
#include <iostream>
#include <map>

using namespace std;

int main() {
    int n;
    cin >> n;

    map<long long, int> note;

    for (int i = 0; i < n; ++i) {
        long long v;
        cin >> v;
        if (note.find(v) == note.end()) {
            note[v] = 1;
        } else {
            note[v]++;
        }
    }

    map<long long, int>::iterator iter = note.begin();
    int max_cnt = 0;
    long long max_v = -(2 << 62)-1;
    while (iter != note.end()) {
        if (iter->second > max_cnt) {
            max_v = iter->first;
            max_cnt = iter->second;
        }
        iter++;
    }

    cout << max_v << '\n';

    return 0;
}